File: src/ObjectManager/Object.Tes/FilePacks/Records/045-ANIO.Animated Object.cs
using OA.Core;

namespace OA.Tes.FilePacks.Records
{
    public class ANIORecord : Record
    {
        public override string ToString() => $"ANIO: {EDID.Value}";
        public STRVField EDID { get; set; } // Editor ID
        public MODLGroup MODL; // Model
        public FMIDField<IDLERecord> DATA; // IDLE animation

        public override bool CreateField(UnityBinaryReader r, GameFormatId formatId, string type, int dataSize)
        {
            switch (type)
            {
                case "EDID": EDID = new STRVField(r, dataSize); return true;
                case "MODL": MODL = new MODLGroup(r, dataSize); return true;
                case "MODB": MODL.MODBField(r, dataSize); return true;
                case "DATA": DATA = new FMIDField<IDLERecord>(r, dataSize); return true;
                default: return false;
            }
        }
    }
}

File: src.mac/ObjectManager/ObjectManager/Object.Tes/FilePacks/Records/045-ANIO.Animated Object.swift
//
//  ANIORecord.swift
//  ObjectManager
//
//  Created by Sky Morey on 5/28/18.
//  Copyright © 2018 Sky Morey. All rights reserved.
//

public class ANIORecord: Record, IHaveEDID, IHaveMODL{
    public override var description: String { return "ANIO: \(EDID)" }
    public var EDID: STRVField = STRVField_empty // Editor ID
    public var MODL: MODLGroup? = nil // Model
    public var DATA: FMIDField<IDLERecord>! // IDLE animation
    
    override func createField(_ r: BinaryReader, for format: GameFormatId, type: String, dataSize: Int) -> Bool {
        switch type {
        case "EDID": EDID = r.readSTRV(dataSize)
        case "MODL": MODL = MODLGroup(r, dataSize)
        case "MODB": MODL!.MODBField(r, dataSize)
        case "DATA": DATA = FMIDField<IDLERecord>(r, dataSize)
        default: return false
        }
        return true
    }
}
